Data/Primitive/PrimArray.hs

```diff
@@ -0,0 +1,938 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE MagicHash #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UnboxedTuples #-}
+
+{-# OPTIONS_GHC -Wall #-}
+
+module Data.Primitive.PrimArray
```

Comment:
There seems to be at least one set of non-standard naming conventions used in this module (e.g., the `-A` and `-P` suffix distinction), so perhaps it would be a good idea to give an overview of them in the module Haddocks.